### PR TITLE
Adding coding standard document

### PR DESCRIPTION
**Pull request by JinendraGhodke**
Ginger coding standard document (Naming conventions, common coding practices and so on)

### CODE REVIEW

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    //Namespace
+    Amdocs.Ginger.Common.Repository;
+
+    //Type
+    public class StreamReader{...}
+
+    //Method
+    public void StartDriver(){...};
+```
+* Do chose readable identifier names e.g. Horizontal alignment is more English readable than alignment horizontal
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: fix typo of chose --> choose

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	    Penny,
+	    Nickel,
+	    Dime,
+	    Quarter,
+	    Dollar
+	}
+```
+
+####XAML Naming conventions
+* Use “x” prefix all XAML control names
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Add use x:Name

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+* Use use the type name as suffix for a xaml control names e.g. Btn for Button.
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Remove Btn - only exact name as suffix

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	    Dime,
+	    Quarter,
+	    Dollar
+	}
+```
+
+####XAML Naming conventions
+* Use “x” prefix all XAML control names
+* Use use the type name as suffix for a xaml control names e.g. Btn for Button. 
+ So name of reset button in xaml will be "xResetBtn"
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: x:Name="xResetButton"

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	    Dollar
+	}
+```
+
+####XAML Naming conventions
+* Use “x” prefix all XAML control names
+* Use use the type name as suffix for a xaml control names e.g. Btn for Button. 
+ So name of reset button in xaml will be "xResetBtn" 
+
+#### Ginger namespace
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: NameSpaces will be in another doc - we can have global ref here to say check NameSpcaes

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+- Amdocs.Ginger.Utils (general standalone methods)
+- Amdocs.Ginger.Report (HTML/mail report)
+- Amdocs.Ginger.Usage (auto log)
+- Amdocs.Ginger.SourceControl
+- Amdocs.Ginger.Workspace (cross dlls sharing cache objects/events)
+
+
+##Common Coding practices
+*	Declare all member variables at the top of a class with static variables at the very top. (Generally accepted practice that prevents the need to hunt for variable declaration)
+*	For iterating an observable list use for loop instead of foreach. Because if list collection is modified during foreach it throws an exception
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Provide code sample

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+*	
+
+
+####Guidelines for Exception handling
+*	Use the Reporter class to show any user message instead of a message box
+```cs
+ Reporter.ToUser(eUserMsgKeys.MissingActivityAppMapping);
```
Comment (yaronweiss) [inline Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: We will create a doc for reporter and message box

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 1 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Handled

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 2 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Handled

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 3 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Handled

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 4 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Handled

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 5 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Added a link reference to the namespace document

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 6 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: Added code sample

File: Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md (hunk 7 above)
Comment (JinendraGhodke) [reply Ginger/DeveloperZone/Documents/General/Ginger Coding Standards.md]: For now keeping it. We will point it to the new document once created

**Comment by yaronweiss** [comment]
Done push and delete branch

**Pull request merged**